evals/api.py

```diff
@@ -234,77 +236,6 @@ def sample_freeform(
         headers={},
         **kwargs,
     )
-    return postprocess_sample_freeform(
-        response,
-        actual_prompt,
-        metadata,
-        model_spec,
-        n_samples=n_samples,
-        return_logprobs=return_logprobs,
-        **kwargs)
-
-
-def postprocess_sample_freeform(
-    response: dict,
-    prompt: Union[OpenAICreatePrompt, OpenAICreateChatPrompt, Prompt],
-    metadata: dict,
-    model_spec: ModelSpec,
-    *,
-    n_samples: Optional[int] = None,
-    return_logprobs: bool = False,
-    **kwargs,
-) -> Union[str, list[str], dict]:
-    """
-    Records the sampled response, prompt and metedata, and returns the sampled text.
-    Typically called after `sample_freeform`.
-
-    ARGS
-    ====
-    `response`: The result of the API call.
-    `prompt`: See `completion_query`.
-    `n_samples`: The number of samples to generate (1 if None).
-    `return_logprobs`: If True, returns the tokens and corresponding logprobs
-        in addition to the sampled text.
-    `kwargs`: See `completion_query`.
-
-    RETURNS
-    =======
-    If `return_logprobs` is True, returns a dict with the sampled text, tokens,
-        and corresponding logprobs. If `n_samples` is None, the outer list is
-        removed from all values.
-    Otherwise, returns the sampled text, or a list of sampled texts if
-        `n_samples` is not None.
-    """
-    sampled = [choice["text"] for choice in response["choices"]]
-    if n_samples is None:
-        sampled = sampled[0]
-    record_sampling(prompt=prompt, sampled=sampled, metadata=metadata)
-
-    if return_logprobs:
-        assert not model_spec.is_chat, "logprobs only works for non-chat models"
-        assert not kwargs.get("logprobs") is None
-
-        def _maybe_tokens(logprobs: Optional[dict]) -> Optional[list[str]]:
-            return logprobs["tokens"] if logprobs is not None else None
-
-        def _maybe_logprobs(logprobs: Optional[dict]) -> Optional[list[float]]:
-            return logprobs["token_logprobs"] if logprobs is not None else None
-
-        def _maybe_top_logprobs(logprobs: Optional[dict]) -> Optional[list[dict[str, float]]]:
-            return [dict(x) for x in logprobs["top_logprobs"]] if logprobs is not None else None
-
-        tokens = [_maybe_tokens(choice["logprobs"]) for choice in response["choices"]]
-        logprobs = [_maybe_logprobs(choice["logprobs"]) for choice in response["choices"]]
-        top_logprobs = [_maybe_top_logprobs(choice["logprobs"]) for choice in response["choices"]]
-        if n_samples is None:
-            tokens = tokens[0]
-            logprobs = logprobs[0]
-            top_logprobs = top_logprobs[0]
-        return {
-            "text": sampled,
-            "tokens": tokens,
-            "logprobs": logprobs,
-            "top_logprobs": top_logprobs,
-        }
-
+    sampled = result.get_completions()[0]
+    record_sampling(prompt=result.prompt, sampled=sampled)
```

Comment:
Should we leave it up to CompletionFn to record the sampling event? It might actually be better since sometimes there could be some implementation-specific information that we might want to record.